### PR TITLE
Add rabbitmq operator

### PR DESCRIPTION
**Pull request by harryge00**

### CODE REVIEW

**Comment by zmalik** [comment]
@harryge00 this is looking good! 
can you please add the operator here https://github.com/kudobuilder/operators/blob/master/kudo-test.yaml#L15 ?
so the CI can test the operatorversion installation at least.

File: repository/rabbitmq/operator/operator.yaml
Comment (zmalik) [inline repository/rabbitmq/operator/operator.yaml]:
```suggestion
url: https://www.rabbitmq.com/
```

File: repository/rabbitmq/operator/operator.yaml
Comment (zmalik) [inline repository/rabbitmq/operator/operator.yaml]: let's use `rabbitmq` name as it's implicit that it will be an KUDO operator
and install cmd will look like `kubectl kudo install rabbitmq` vs `kubectl kudo install rabbitmq-operator`

File: repository/rabbitmq/operator/templates/role.yaml
Comment (zmalik) [inline repository/rabbitmq/operator/templates/role.yaml]: remove this?

**Review by zmalik — CHANGES_REQUESTED** [review]
thanks you @harryge00 !
left a few comments

**Comment by harryge00** [comment]
@zmalik Thanks for the review. I have resolved the comments.

File: repository/rabbitmq/operator/operator.yaml
Comment (zmalik) [inline repository/rabbitmq/operator/operator.yaml]: @harryge00 can you change the version to `operatorVersion`? CI is failing because of that as we moved to KUDO `0.10.0`
and add the `appVersion` also which would be `3.8` in this case

**Review by zmalik — CHANGES_REQUESTED** [review]

**Review by zmalik — APPROVED** [review]
LGTM for an initial version of rabbitmq operator
@harryge00 let's wait for another review

**Comment by zmalik** [comment]
let's merge this and we can iterate on this
Ideally, if you can add some simple tests, later would help with maintaining this operator here 
you can check harness test examples from Kafka: https://github.com/kudobuilder/operators/tree/master/repository/kafka/tests/kafka-upgrade-test

**Comment by zmalik** [comment]
rabbitmq operator `3.8.0_0.1.0` published 
```
$ kubectl kudo install rabbitmq
operator.kudo.dev/v1beta1/rabbitmq created
operatorversion.kudo.dev/v1beta1/rabbitmq-0.1.0 created
instance.kudo.dev/v1beta1/rabbitmq-instance created
```
```
$ kubectl kudo plan status  --instance=rabbitmq-instance
Plan(s) for "rabbitmq-instance" in namespace "default":
.
└── rabbitmq-instance (Operator-Version: "rabbitmq-0.1.0" Active-Plan: "deploy")
    └── Plan deploy (serial strategy) [COMPLETE]
        └── Phase main (parallel strategy) [COMPLETE]
            └── Step everything [COMPLETE]
```

```
$ kubectl get pods
NAME                           READY   STATUS    RESTARTS   AGE
rabbitmq-instance-rabbitmq-0   1/1     Running   0          42s
rabbitmq-instance-rabbitmq-1   1/1     Running   0          28s
rabbitmq-instance-rabbitmq-2   1/1     Running   0          23s
```

**Pull request merged**
